File: offer/02-替换空格.cpp
#include <iostream>
#include <string>
using namespace std;

string replace(string  str) {
    string s;
    for (int i = 0; i < str.length(); ++i) {
        if (str[i] == ' ') {
            s += "%20";
            continue;
        }
        s += str[i];
    }
    return s;
}

File: offer/go/02-替换空格.go
package offer

func Replace(str string) string {
	var s string = ""
	for i := 0; i < len(str); i++ {
		if str[i] == ' ' {
			s = s + "%20"
			continue
		}
		s = s + string(str[i])
	}
	return s
}
